src/app.js

```diff
@@ -181,10 +223,65 @@ gpii.app.exit = function (that) {
 };
 
 /**
-  * Handles when a user token is keyed out through other means besides the task tray key out feature.
-  * @param that {Component} An instance of gpii.app
-  * @param keyedOutUserToken {String} The token that was keyed out.
-  */
+ * Listens to any changes in the lifecycle managers logonChange model and calls the
+ * appropriate functions in gpii-app for notifying of the current user state
+ */
+gpii.app.logonChangeListener = function (that, model) {
+    if (model.inProgress === true) {
+        gpii.app.displayWaitDialog(that);
+    } else {
+        gpii.app.dismissWaitDialog(that);
+    }
+};
+
+/**
+ * Shows the dialog on users screen with the message passed as parameter.
+ * Records the time it was shown in `dialogStartTime` which we need when
+ * dismissing it (checking whether it's been displayed for the minimum amount of time)
+ *
+ * @param that {Object} the app module
+ * @param msg {String} the message to display to the user
+ */
+gpii.app.displayWaitDialog = function (that) {
+    that.dialog.show();
+    // Hack to ensure it stays on top, even as the GPII autoconfiguration starts applications, etc., that might
+    // otherwise want to be on top
+    // see amongst other: https://blogs.msdn.microsoft.com/oldnewthing/20110310-00/?p=11253/
+    // and https://github.com/electron/electron/issues/2097
+    var interval = setInterval(function () {
+        if (!that.dialog.isVisible()) {
+            clearInterval(interval);
+        };
+        that.dialog.setAlwaysOnTop(true);
+    }, 100);
+
+    that.dialogStartTime = Date.now();
+};
+
+/**
+ * Dismisses the dialog. If less than `that.dialogMinDisplayTime` ms have passed since we first displayed
+ * the window, the function waits until `dialogMinDisplayTime` has passed before dismissing it.
+ *
+ * @param that {Object}: the app
```

Comment:
no ":" after the brackets - according to our undocumented and unused parsing rules for the auto documentation